tracing-tracy/src/lib.rs

```diff
@@ -291,3 +351,117 @@ fn main() {
         tests::test();
     }
 }
+
+mod utils {
+    use crate::MAX_CACHE_SIZE;
+    use std::cell::{Cell, UnsafeCell};
+    use std::mem;
+    use std::mem::ManuallyDrop;
+    use std::ops::{Deref, DerefMut};
+    use std::sync::atomic::Ordering;
+
+    pub struct VecCell<T>(UnsafeCell<Vec<T>>);
+
+    impl<T> VecCell<T> {
+        pub const fn new() -> Self {
+            Self(UnsafeCell::new(Vec::new()))
+        }
+
+        pub fn push(&self, item: T) {
+            unsafe { &mut *self.0.get() }.push(item);
```

Comment:
Please write down safety comments (i.e. why this unsafe operation is guaranteed to be correct / how are the preconditions imposed by the callee are being satisfied.)

Reply:
Done.